errors.go

```diff
@@ -18,3 +18,38 @@ func (e errorList) Error() string {
 	}
 	return "multiple errors:\n" + strings.Join(msgs, "\n")
 }
+
+// IgnorableError is an error that has an extra method to tell whether it should
+// be ignored. Ignored errors may be omitted from standard program output.
+type IgnorableError interface {
+	error
+	Ignore() bool
+}
+
+// ignoredError implements IgnorableError, always returning true.
+type ignoredError struct {
+	err error
+}
+
+func (e *ignoredError) Error() string {
+	if e.err == nil {
+		return ""
+	}
+	return e.err.Error()
+}
+
+func (e *ignoredError) Ignore() bool {
+	return true
+}
+
+var _ IgnorableError = (*ignoredError)(nil)
```

Comment:
The stdlib has even more examples:

https://github.com/golang/go/blob/master/src/encoding/json/stream.go#L263-L264
https://github.com/golang/go/blob/master/src/io/multi.go#L64
https://github.com/golang/go/blob/master/src/net/http/server.go#L1443
https://github.com/golang/go/blob/master/src/testing/testing.go#L382-L383

```
$ grep -Er 'var _ .*\(nil\)$' `go env GOROOT`/src
/usr/local/go/src/io/multi.go:var _ stringWriter = (*multiWriter)(nil)
/usr/local/go/src/encoding/json/stream.go:var _ Marshaler = (*RawMessage)(nil)
/usr/local/go/src/encoding/json/stream.go:var _ Unmarshaler = (*RawMessage)(nil)
/usr/local/go/src/encoding/json/decode_test.go:var _ encoding.TextUnmarshaler = (*unmarshalerText)(nil)
/usr/local/go/src/encoding/json/decode_test.go:var _ encoding.TextUnmarshaler = (*u8marshal)(nil)
/usr/local/go/src/encoding/xml/marshal_test.go:var _ Marshaler = (*MyMarshalerTest)(nil)
/usr/local/go/src/encoding/xml/marshal_test.go:var _ MarshalerAttr = (*MyMarshalerAttrTest)(nil)
/usr/local/go/src/encoding/xml/read_test.go:var _ Unmarshaler = (*MyCharData)(nil)
/usr/local/go/src/encoding/xml/read_test.go:var _ UnmarshalerAttr = (*MyAttr)(nil)
/usr/local/go/src/net/http/server.go:var _ closeWriter = (*net.TCPConn)(nil)
/usr/local/go/src/testing/testing.go:var _ TB = (*T)(nil)
/usr/local/go/src/testing/testing.go:var _ TB = (*B)(nil)
/usr/local/go/src/crypto/aes/ctr_s390x.go:var _ ctrAble = (*aesCipherAsm)(nil)
/usr/local/go/src/crypto/aes/cbc_s390x.go:var _ cbcEncAble = (*aesCipherAsm)(nil)
/usr/local/go/src/crypto/aes/cbc_s390x.go:var _ cbcDecAble = (*aesCipherAsm)(nil)
/usr/local/go/src/crypto/aes/aes_gcm.go:var _ gcmAble = (*aesCipherGCM)(nil)
```

:smile: